src/Invitation.ts

```diff
@@ -67,6 +74,148 @@ export class InvitationV1 implements invitation.InvitationV1 {
+      )
+    }
+    this._sendKeyBundle = new SignedPublicKeyBundleV2(sendKeyBundle)
+    this._inboxKeyBundle = new SignedPublicKeyBundleV2(inboxKeyBundle)
+    this._createdNs = createdNs
+  }
+
+  public get createdNs(): Long {
+    return this._createdNs
+  }
+
+  public get sendKeyBundle(): SignedPublicKeyBundleV2 {
+    return this._sendKeyBundle
+  }
+
+  public get inboxKeyBundle(): SignedPublicKeyBundleV2 {
+    return this._inboxKeyBundle
+  }
+
+  public async getPeerAddress(selfAddress: string): Promise<string> {
+    if (this._peerAddress) return this._peerAddress
+    const senderAddress = this.sendKeyBundle.getLinkedAddress(
+      AccountLinkedRole.SEND_KEY
+    )
+    const isSender = senderAddress === selfAddress
+    if (isSender && this.selfHeader) {
+      this._peerAddress = this.selfHeader.recipientAddress
+    } else if (!isSender && this.peerHeader) {
+      this._peerAddress = senderAddress
+    } else {
+      throw new Error('SealedInvitationHeaderV2 has wrong peer or self header')
+    }
+    return this._peerAddress
+  }
+
+  toBytes(): Uint8Array {
+    return invitation.SealedInvitationHeaderV2.encode(this).finish()
+  }
+
+  static fromBytes(bytes: Uint8Array): SealedInvitationHeaderV2 {
+    return new SealedInvitationHeaderV2(
+      invitation.SealedInvitationHeaderV2.decode(bytes)
+    )
+  }
+}
+
+export class SealedInvitationV2 implements invitation.SealedInvitationV2 {
+  headerBytes: Uint8Array
+  ciphertext: Ciphertext
+  private _header?: SealedInvitationHeaderV2
+  private _invitation?: InvitationV1
+
+  constructor({ headerBytes, ciphertext }: invitation.SealedInvitationV2) {
+    if (!headerBytes || !headerBytes.length) {
+      throw new Error('Missing header bytes')
+    }
+    if (!ciphertext) {
+      throw new Error('Missing ciphertext')
+    }
+    this.headerBytes = headerBytes
+    this.ciphertext = new Ciphertext(ciphertext)
+  }
+
+  /**
+   * Accessor method for the full header object
+   */
+  get header(): SealedInvitationHeaderV2 {
+    // Use cached value if already exists
+    if (this._header) {
+      return this._header
+    }
+    this._header = SealedInvitationHeaderV2.fromBytes(this.headerBytes)
+    return this._header
+  }
+
+  /**
+   * getInvitation decrypts and returns the InvitationV1 stored in the ciphertext of the Sealed Invitation
+   */
+  async getInvitation(
+    viewerInboxKeyBundle: PrivateKeyBundleV3
+  ): Promise<InvitationV1> {
+    // Use cached value if already exists
+    if (this._invitation) {
+      return this._invitation
+    }
+    const header = this.header
+    if (
+      !viewerInboxKeyBundle.accountLinkedKey.matches(
+        header.inboxKeyBundle.accountLinkedKey
+      )
+    ) {
+      throw new Error('Inbox key bundle does not match invite')
+    }
+    const secret = await viewerInboxKeyBundle.sharedSecret(
+      header.sendKeyBundle,
+      header.inboxKeyBundle.preKey,
+      true /* isRecipient */
```

Comment:
Might be worth adding some comments explaining how `SealedInvitationV2` is never used for self-invites and that's why you can hardcode to this value to `true`